### PR TITLE
fix(refactor): component viewer UI

### PR DESCRIPTION
**Pull request by Fupete**
fix: #871 and do a lot more...

We need to simplify the UI of the component viewer. 

Url paths to test the result with various settings: 
- `/design-system/fondamenti/tipografia` for multiple viewers with code on page
- `/design-system/componenti/accordion`, first tab without code and third tabs with all on

To do:
- [x] Extract focus mode icon, usefull also for a11y testing
- [x] Enhance a11y of responsive buttons group: 
  - [x] Move label to aria
  - [x] Convert to group
  - [x] Fix role="group"
- [x] Reorder elements
  - [x] Move responsive buttons and focus mode icon before the the preview
  - [x] Move "Mostra codice a capo" inside code viewer
 
----
Current: 
immagine

----
Progress: 
immagine

### CODE REVIEW

**Review by zetareticoli — COMMENTED** [review]
I see two issues with the new layout:
- position of viewport buttons creates a new line between variants select and component preview; could we improve this, moving it on the same row?
- the icon near to viewport buttons isn't clear, I don't know what happens when clicking on it; a sort of "external" link could work better
empty-space

**Comment by Fupete** [comment]
@zetareticoli tx for the feedback. Answers inline. 

>     * position of the viewport buttons creates a new line between variants select and component preview; could we improve this, by moving it to the same row?

Not really, at the moment they are in two different and complex React components. This could be a future improvement. 

>     * the icon near to viewport buttons isn't clear, I don't know what happens when clicking on it; a sort of "external" link could work better

It is the current icon. I am not convinced about the external icon, we already use it for the Bootstrap Italia documentation link. And above all, the function activated by the icon (open in a new window) is something we have been talking about in tutorials and webinars for a long time as a 'focus mode' (e.g. to test a11y on components without noise). 

>     * the "100%" label is unclear, what does it mean? We could have a 100% layout on a smartphone too

This is currently a test. The "desktop" label is somewhat wrong indeed, in fact it is a "fullwidth" preview that depends on which device you are viewing the page and what maximum resolution you are viewing. In reality, it is only "100%" like Orbit UI already use for example.
immagine

**Comment by Fupete** [comment]
@zetareticoli 
- Ok for the new icon. We can substitute as soon as ready. 
- As preview resolutions I want to stay with mobile/tablet/desktop approach. They are not the media queries or our UI Kit "cases". They are a preview of how it renders in different context. We need the three steps IMHO.

**Comment by zetareticoli** [comment]
I was looking at how PatternLab manages similar icons and labels:
Screenshot 2023-11-03 alle 19 45 36


Screenshot 2023-11-03 alle 19 46 35

It makes the device name coincide with that of the breakpoint

**Comment by zetareticoli** [comment]
Anyway, "full" works better than "100%". 

You can have a component with `width: 100%` inside a `300px` fixed-size container. It will look like the mobile version.

**Comment by Fupete** [comment]
@zetareticoli wdyt? We can work on better icons later. 

In the meantime, I would like to finish and merge this, and then proceed with a new issue/PR pair to improve the display of the viewer on a "componente" page. The idea there is to try moving the preview buttons to the right of the variant <select>. A rather large code refactor would be required... so I would prefer to separate it from this PR and keep it as a separate task.

**Review by zetareticoli — APPROVED** [review]

File: src/components/component-view/component-view.js [hunk 1]
```diff
@@ -147,9 +147,20 @@ function ComponentView({
...
-    "border-bottom d-flex flex-column align-items-center",
-    { "pb-4": responsiveButtonsItems },
+  const componentStyles = classNames("d-flex flex-column align-items-center", {
+    "me-lg-4 me-xl-5": componentViewerData?.variants,
+  });
+
+  const accordionHeadingStyle = classNames(
+    "d-flex justify-content-between align-items-center",
+    {},
+  );
```
Comment (bfabio) [inline src/components/component-view/component-view.js:157]:
```suggestion
  const accordionHeadingStyle = "d-flex justify-content-between align-items-center";
```

**Review by bfabio — APPROVED** [review]

**Pull request merged**